src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerExpediteFinalOrder.java

```diff
@@ -0,0 +1,192 @@
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.JUDGE;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@Component
+public class CaseworkerExpediteFinalOrder implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_EXPEDITE_FINAL_ORDER = "caseworker-expedite-final-order";
+
+    @Autowired
+    private NotificationDispatcher notificationDispatcher;
+    
+    @Autowired
+    private FinalOrderGrantedNotification finalOrderGrantedNotification;
+
+    @Autowired
+    private GenerateFinalOrderCoverLetter generateFinalOrderCoverLetter;
+
+    @Autowired
+    private GenerateFinalOrder generateFinalOrder;
+
+    @Autowired
+    private Clock clock;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_EXPEDITE_FINAL_ORDER)
+            .forStates(FinalOrderRequested, FinalOrderPending, GeneralConsiderationComplete)
+            .name("Expedite Final order")
+            .description("Expedite Final order")
+            .showSummary()
+            .showEventNotes()
+            .endButtonLabel("Submit")
+            .aboutToStartCallback(this::aboutToStart)
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .submittedCallback(this::submitted)
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grantHistoryOnly(SOLICITOR, SUPER_USER, LEGAL_ADVISOR, JUDGE))
+            .page("expediteFinalOrder")
+            .pageLabel("Expedite Final Order")
+            .complex(CaseData::getDocuments)
+                .mandatory(CaseDocuments::getGeneralOrderDocumentNames)
+            .done()
+            .complex(CaseData::getFinalOrder)
+                .complex(FinalOrder::getExpeditedFinalOrderAuthorisation)
+                    .mandatory(ExpeditedFinalOrderAuthorisation::getExpeditedFinalOrderJudgeName)
+                .done()
+                .mandatory(FinalOrder::getGranted)
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(CaseDetails<CaseData, State> details) {
+        log.info("{} about to start callback invoked for Case Id: {}", CASEWORKER_EXPEDITE_FINAL_ORDER, details.getId());
+        var caseData = details.getData();
+
+        if (caseData.getGeneralOrders() == null) {
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .errors(Collections.singletonList("No general order documents found.  Unable to continue."))
+                .build();
+        }
+
+        final var generalOrderDocuments = caseData.getGeneralOrders()
```

Comment:
if the documents are null, do we want them accessing this event? Would it be better to add the check for null in the initial check of getGeneralOrders() == null above. As then we can assume if they get past that point documents do exist.

Reply:
Not sure I'm following this - we are not concerned with the documents field at all for this event.  Can we discuss please?